backend/src/managers/PostManager.ts

```diff
@@ -244,12 +245,20 @@ export default class PostManager {
         const commentRaw = await this.commentRepository.createComment(userId, postId, parentCommentId, content, language, parseResult.text);
         this.userManager.clearUserRestrictionsCache(userId);
 
-        for (const mention of parseResult.mentions) {
+        let parentAuthor: UserInfo | undefined;
+        if (parentCommentId) {
+            const parentComment = await this.commentRepository.getComment(parentCommentId);
+            parentAuthor = await this.userManager.getById(parentComment.author_id);
+        }
+        for (const mention of new Set(parseResult.mentions)) {
+            if (mention.toLowerCase() === parentAuthor?.username.toLowerCase()) {
+                continue;
+            }
             await this.notificationManager.sendMentionNotify(mention, userId, postId, commentRaw.comment_id);
         }
 
-        if (parentCommentId) {
-            await this.notificationManager.sendAnswerNotify(parentCommentId, userId, postId, commentRaw.comment_id);
+        if (parentAuthor) {
+            await this.notificationManager.sendAnswerNotify(parentAuthor.id, userId, postId, commentRaw.comment_id);
```

Comment:
Previously this was commentId and now we are sending userId - is that correct?

Reply:
Yep, the signature has changed (part of the logic was extracted ouside).